sources/HUBViewModelDiff.m

```diff
@@ -57,7 +64,9 @@ + (instancetype)diffFromViewModel:(id<HUBViewModel>)fromViewModel
                         algorithm:(HUBDiffAlgorithm)algorithm
 {
     NSParameterAssert(algorithm);
-    return algorithm(fromViewModel, toViewModel);
+    HUBViewModelDiff *diff =  algorithm(fromViewModel, toViewModel);
+    [diff calculateHeaderChangesFromViewModel:fromViewModel toViewModel:toViewModel];
+    return diff;
```

Comment:
Yeah it would be implemented exactly the same way, only moved to a different place :smile:

I'm only arguing that since the prototype of the diffing function says _"Take these two view models and return me a complete list of changes between them"_, that should also include whether or not the header changed, as it is a part of the view model. If we modify the diff after the algorithm has run, then the algorithms no longer deliver a complete diff.

In addition to that, if we would ever want to change the header diffing, (e.g. by differentiating between a completely new header or in-place updates to an existing header), then this could differ between algorithms, as the header can have list of child components, much like the view model's body components.